presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java

```diff
@@ -149,14 +151,21 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
                 throw new PrestoException(NO_NODES_AVAILABLE, "No nodes available to run query");
             }
 
-            InternalNode chosenNode = chooseLeastBusyNode(candidateNodes, assignmentStats::getTotalSplitCount, preferredNodeCount, maxSplitsPerNode);
-            if (chosenNode == null) {
+            Optional<NodeResult> chosenNode = chooseLeastBusyNode(candidateNodes, assignmentStats::getTotalSplitCount, preferredNodeCount, maxSplitsPerNode);
+            if (!chosenNode.isPresent()) {
                 chosenNode = chooseLeastBusyNode(candidateNodes, assignmentStats::getQueuedSplitCountForStage, preferredNodeCount, maxPendingSplitsPerTask);
             }
 
-            if (chosenNode != null) {
-                assignment.put(chosenNode, split);
-                assignmentStats.addAssignedSplit(chosenNode);
+            if (chosenNode.isPresent()) {
+                split = new Split(
+                        split.getConnectorId(),
+                        split.getTransactionHandle(),
+                        split.getConnectorSplit(),
+                        split.getLifespan(),
+                        new SplitContext(chosenNode.get().isCacheable()));
+
+                assignment.put(chosenNode.get().getInternalNode(), split);
```

Comment:
nit: not sure how much value it would add, but maybe have a variable for `chosenNode.get().getInternalNode()` since it is used twice.